src/index.js

```diff
@@ -80,20 +83,38 @@ class SpriteAnimator extends Component {
       return this.loadSprite()
     }
 
-    const {shouldAnimate, timeout, stopLastFrame, onEnd} = this.props
+    const {shouldAnimate, fps, stopLastFrame, onEnd} = this.props
     if (shouldAnimate) {
       const {maxFrames, currentFrame} = this.state
       const nextFrame = currentFrame + 1 >= maxFrames ? 0 : currentFrame + 1
+
+      if (!shouldAnimate) {
+        return
+      }
       if (nextFrame === 0 && stopLastFrame) {
         return onEnd()
       }
-      setTimeout(() => {
-        if (!this.props.shouldAnimate) return
-        this.setState({currentFrame: nextFrame})
-      }, timeout)
+      this.interval = 1000 / fps
+
+      this.animationId = raf(time => this.animate(nextFrame, time))
     }
   }
 
+  animate (nextFrame, time) {
+    if (!this.prevTime) {
+      this.prevTime = time
+    }
+
+    const delta = time - this.prevTime
+    if (delta < this.interval) {
+      this.animationId = raf(time => this.animate(nextFrame, time))
+      return
```

Comment:
bail out if the animation is faster than the specified framerate, check again in the next animation frame. this starts an inner loop. the outer loop componentDidUpdate -> animate -> setState -> componentDidUpdate -> ... is not affected by this, we just "wait" inside of animate until the target framerate is matched.